test/Integration/SqlTriggerBindingIntegrationTests.cs

```diff
@@ -163,5 +207,36 @@ private void DeleteProducts(int first_id, int last_id)
                 id += 1;
             }
         }
+
+        private async Task<string> AssertTriggerListenerError(string functionName)
+        {
+            string errorMessage = null;
+            var tcs = new TaskCompletionSource<bool>();
+
+            void OutputHandler(object sender, DataReceivedEventArgs e)
+            {
+                if (errorMessage == null && e.Data?.Contains("Failed to start SQL trigger listener") == true)
+                {
+                    string exceptionPrefix = "Exception: System.InvalidOperationException: ";
+                    int index = e.Data.IndexOf(exceptionPrefix, StringComparison.Ordinal);
+                    Assert.NotEqual(-1, index);
+                    errorMessage = e.Data[(index + exceptionPrefix.Length)..];
+                    tcs.SetResult(true);
+                }
+            };
+
+            this.StartFunctionHost(functionName, Common.SupportedLanguages.CSharp, false, OutputHandler);
+            Task completedTask = await Task.WhenAny(tcs.Task, Task.Delay(TimeSpan.FromSeconds(5)));
+            Assert.Equal(tcs.Task, completedTask);
+            this.FunctionHost.OutputDataReceived -= OutputHandler;
+
+            // WebJobs SDK retries forever to start the trigger-listener until it succeeds, which makes the Functions
+            // Host process never exit by itself in case of an error.
+            Assert.False(this.FunctionHost.HasExited);
```

Comment:
`Assert.True` and `Assert.False` allow a message parameter, please use that to give a descriptive failure message.

But also - why do we need this assert? If the process exits (maybe the host changes its behavior) then do we really care? We can just kill the host regardless so this assert doesn't seem useful.

Reply:
It is not much useful. I can remove it. If I remember correctly, the functions host stops and does not retry and exit if there is an error in early stage, for example, when requesting a trigger-binding from binding provider, etc.